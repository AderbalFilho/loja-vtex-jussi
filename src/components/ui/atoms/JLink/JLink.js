import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './JLink.scss';

function JLink({ children, classes, ...rest }) {
  const className = classNames('j-link', classes);

  return (
    <a className={className} {...rest}>
      {children}
    </a>
  );
}

JLink.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
};

export default JLink;
