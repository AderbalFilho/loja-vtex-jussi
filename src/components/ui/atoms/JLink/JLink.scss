@import 'styles/mixin';

.j-link {
  @extend %text-paragraph;

  color: var(--black);
  text-decoration: none;

  &:hover {
    text-decoration: underline solid var(--pink-jussi) 0.2rem;
  }
}
