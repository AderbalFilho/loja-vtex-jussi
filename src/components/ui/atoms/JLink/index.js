import JLink from './JLink';

export default JLink;
