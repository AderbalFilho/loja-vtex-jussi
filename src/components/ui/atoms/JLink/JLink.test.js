import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import JLink from './JLink';

describe('JLink', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<JLink>Jüssi</JLink>);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  it('renders children', () => {
    const { container } = render(<JLink>Jüssi</JLink>);
    const link = container.querySelector('.j-link');

    expect(link.textContent).toBe('Jüssi');
  });

  it('renders classes', () => {
    const { container } = render(<JLink classes="nav-link">Jüssi</JLink>);
    const link = container.querySelector('.j-link');

    expect(link.classList).toContain('nav-link');
  });

  it('renders rest', () => {
    const { container } = render(<JLink href="/search">Jüssi</JLink>);
    const link = container.querySelector('.j-link');

    expect(link.getAttribute('href')).toBe('/search');
  });
});
