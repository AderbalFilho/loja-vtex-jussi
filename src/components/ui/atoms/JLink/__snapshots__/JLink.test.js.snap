// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JLink matches snapshot 1`] = `
<a
  className="j-link"
>
  JÃ¼ssi
</a>
`;
